src/liquid/match_visitor.cr

```diff
@@ -0,0 +1,59 @@
+require "./blocks"
+
+module Liquid
+  class MatchVisitor < Visitor
+    def visit(node : Node) : String
```

Comment:
```suggestion
    @io = IO::Memory.new
    
    def visit(node : Node) : String
```

Would be faster if you create a `IO::Memory` object in the constructor then add the contents to it on each visitor instead of allocating a lot of small string and concatenate them in a larger one few times.

Reply:
In the cases of loops and conditionals, they don't just need to be added, they need to be joined with `(` and `|`. How would I be able to do that if I used your method?

Reply:
It's probably only marginally faster, or imperceptibly faster. This would be worth doing as a test to speed test both versions I think. I'm all for performance optimizations out of the box, but let's do that as another PR if it becomes a noticeable bottleneck. Sound acceptible @hugopl?

Reply:
yeah, it can be done later.